### PR TITLE
[Manifest] Cache - MKP deployment

### PR DESCRIPTION
**Pull request by rui5i**
This PR re-create mkp manifests for cache service after resolving mpdev verify error.
Using Deployment for cache-deployer in both standalone and mkp manifests to prevent uninstall error.

Part of #2904

### CODE REVIEW

**Comment by rui5i** [comment]
/assign @rmgogogo @Bobgy @Ark-kun 
After some bug fix in cache server I recreate this PR for cache mkp manifests. Also changing cache-deployer type from StatefulSet to Deployment. As discussion before, Job has issue in upgrading process and StatefulSet has issue during uninstall when running mpdev verify. And this time mkp test passes as here:  https://pantheon.corp.google.com/cloud-build/builds/0780e86c-a267-41ba-b9cb-b3a9509edd25;step=1;tab=detail?organizationId=433637338589&project=ml-pipeline-test

File: manifests/kustomize/base/cache-deployer/cache-deployer-deployment.yaml [hunk 1]
```diff
@@ -1,12 +1,11 @@
 apiVersion: apps/v1
-kind: StatefulSet
+kind: Deployment
```
Comment (IronPan) [inline manifests/kustomize/base/cache-deployer/cache-deployer-deployment.yaml:2]: can we update the lightweight deployment to keep them in sync

**Comment by rmgogogo** [comment]
double check: tested via kubectl and mpdev both, right?

**Comment by numerology** [comment]
> double check: tested via kubectl and mpdev both, right?

I tested manually via both. Deletion works fine. 

QQ: is there a convenient way already to test upgrade in MKP?

**Comment by rmgogogo** [comment]
To Jiaxiao's questions: https://github.com/kubeflow/pipelines/pull/3438

/lgtm
/approve

Thanks Rui to finish so many in so short time period.

**Pull request merged**